### DISCUSSION TITLE
Jzignet: Zig library to connect Janet and Zig together

### DISCUSSION THREAD

**Discussion by greenfork** [janet-lang/janet#878] [Show and tell]
Do you love Janet but afraid of touching C or C++ to further extend it or embed it? Maybe [Zig](https://ziglang.org/) is a better option!

You can:
* Embed Janet programs into Zig
* Write Janet modules in Zig
* Write bindings in Zig for a C library to be used as a Janet module

Bindings are not complete 100% but all the generally useful things are there. If you need any specific part of the API, feel free to contribute or just ask (and you shall receive).

https://github.com/greenfork/jzignet

**Comment by greenfork** [comment]
0.3.0 release, major overhaul of the library so it is more usable https://github.com/greenfork/jzignet/releases/tag/0.3.0

Also includes template repository for building Zig modules https://github.com/greenfork/jzignet-module-template

**Comment by sogaiu** [comment]
Was able to get the embed_janet example working with 0.3.0 :)

**Reply by greenfork** [reply]
Great! Thank you for checking it out!

**Comment by sogaiu** [comment]
May be there are some interesting bits that might be made available to Janet:

* [awesome-zig](https://github.com/nrdmn/awesome-zig)
* [Zig libraries](https://wiki.nikitavoloboev.xyz/programming-languages/zig/zig-libraries)
